ro_help/hub/views.py

```diff
@@ -180,12 +186,22 @@ def search(self, queryset):
 
     def get_queryset(self):
         needs = self.search(self.get_needs())
-
         return needs.filter(**{name: self.request.GET[name] for name in self.allow_filters if name in self.request.GET})
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         needs = self.search(self.get_needs())
+        # We need a filter for the Red Cross in order to use it for the red
+        # banner.
+        red_cross_filter = needs.filter(ngo__name=RED_CROSS_NAME)
+        if red_cross_filter:
```

Comment:
I think that this `if` is always True since `filter` returns a queryset.

Let's move the entire logic into a separate function called `red_cross` or `fetch_red_cross` or `get_red_cross`.
We can also cache the result in order to avoid unnecessary queries.

```python
context = {**context, **self.red_cross()}
...

def red_cross(self):
    # TODO: cache result

    context = {}

    red_cross = NGO.objects.filter(name=settings.RED_CROS_NAME).first()
    if not red_cross:
        return context

    context["red_cross"] = red_cross

    need = NGONeed.objects.filter(ngo=red_cross, kind=KIND.MONEY).first()
    if not need:
       return context

    context["red_cross_need"] = need
    
    return context
```